dialect/sql/builder.go

```diff
@@ -3592,6 +3621,7 @@ const (
 	OpIn                // IN
 	OpNotIn             // NOT IN
 	OpLike              // LIKE
+	OpILike             // ILIKE
```

Comment:
I'd remove this operator as it's not supported by MySQL/MariaDB, and implement it like [ContainsFold](https://github.com/ent/ent/blob/master/dialect/sql/builder.go#L1803-L1824) predicate - the code can be shared between HasPrefixFold/HasSuffixFold/ContainsFold

Reply:
Okay. let me go through that and adjust accordingly.

Reply:
Adjusted - now the `ContainsFold` relies on a utility method called `escapedLikeFold` which is reused in `HasPrefixFold`, `HasSuffixFold` and `ContainsFold`